subsys/bluetooth/mesh/vnd/silvair_enocean_srv.c

```diff
@@ -335,6 +380,55 @@ const struct bt_mesh_model_op _bt_mesh_silvair_enocean_srv_op[] = {
 	BT_MESH_MODEL_OP_END,
 };
 
+static void delay_update(struct bt_mesh_model *model, uint32_t offset)
+{
+	uint32_t delay;
+
+	delay = delay_get(model->pub, BT_MESH_PUB_MSG_NUM(model->pub));
+	model->pub->msg->data[offset] = model_delay_encode(delay);
+}
+
+static int onoff_update_handler(struct bt_mesh_model *model)
+{
+	struct bt_mesh_onoff_cli *cli = model->user_data;
+
+	if (cli->pub_buf.len != 6 ||
+	    !bt_mesh_model_pub_is_retransmission(model)) {
+		return -EINVAL;
```

Comment:
This is to prevent publication if the periodic publication is enabled. When the publication is triggered from API, the update callback is not called for the first time and only called for retransmissions.

In last transmission, this check is passed, but delay does not change since button state is idle.